conclave-host/src/main/kotlin/com/r3/conclave/host/Gramine.kt

```diff
@@ -0,0 +1,116 @@
+package com.r3.conclave.host
+
+import java.io.File
+import java.io.FileOutputStream
+import java.io.OutputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.concurrent.TimeUnit
+import kotlin.io.path.absolutePathString
+
+object Gramine {
+
+    private lateinit var processGramineDirect: Process
+
+    fun start() {
+        try {
+            val gramineWorkingDirPath = Files.createTempDirectory("conclave-gramine-runtime")
+            generateManifestFile(gramineWorkingDirPath)
+
+            processGramineDirect = ProcessBuilder()
+                .inheritIO()
+                .directory(File(gramineWorkingDirPath.toUri()))
+                .command("gramine-direct", "bash", "-c", "echo \"Gramine bash \'enclave\' started\" && sleep 10000")
+                .start()
+        } catch (exception: Exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    fun stop() {
+        processGramineDirect.destroy()
+        processGramineDirect.waitFor(10L, TimeUnit.SECONDS)
+        if (processGramineDirect.isAlive) {
+            processGramineDirect.destroyForcibly()
+        }
+    }
+
+    fun isGramineEnabled(): Boolean {
+        val envValue = System.getenv("CONCLAVE_GRAMINE_ENABLE") ?: return false
+        return envValue.uppercase() == "TRUE"
+    }
+
+    private fun generateManifestFile(destination: Path) {
+        val file = File(destination.absolutePathString() + "/" + "bash.manifest")
+        writeToFile(manifestContent.toByteArray(), file)
+    }
+
+    private fun writeToFile(content: ByteArray, file: File) {
```

Comment:
I think we should try to use Path API where possible instead of java.io.File

Reply:
What would the issue be if File is used? The file will not exist if an object of type Path is used.

Reply:
I meant in general in this file. I believe we are just trying to use new Path API in general, instead of File API.